### PR TITLE
Add emscripten_console_trace and emscripten_dbg_backtrace. NFC

### PR DESCRIPTION
**Pull request by sbc100**
These are very useful for debugging.  The former simply maps directly to `console.trace`.  The latter use `dbg()` so it writes directly to stderr under node (better for multi-threaded apps).

### CODE REVIEW

File: src/library.js [hunk 1]
```diff
@@ -2090,6 +2094,13 @@ addToLibrary({
     console.error(UTF8ToString(str));
   },
 
+  emscripten_console_trace: (str) => {
+#if ASSERTIONS
+    assert(typeof str == 'number');
```
Comment (dschuff) [inline src/library.js:2099]: wait... why is it called 'str' if it's supposed to be a number? it seems to work fine with regular strings in node?

Comment (sbc100) [reply src/library.js:2099]: These are C APIs, so this is a C string, AKA a pointer, AKA a number.

Comment (dschuff) [reply src/library.js:2099]: Ah of course.

**Review by dschuff — APPROVED** [review]

File: ChangeLog.md [hunk 2]
```diff
@@ -22,6 +22,10 @@ See docs/process.md for more on how version tagging works.
 -----------------------
 - Added support for WebGL extensions EXT_clip_control, EXT_depth_clamp,
   EXT_polygon_offset_clamp and WEBGL_polygon_mode (#20841)
+- New `emscripten_console_trace` and `emscripten_dbg_backtrace` APIs we were
+  added to `console.h`.  The former simply maps directly to `console.trace`.
+  The latter use `dbg()` so it writes directly to stderr under node (better for
```
Comment (kripken) [inline ChangeLog.md:27]:
```suggestion
  The latter uses `dbg()` so it writes directly to stderr under node (better for
```

**Review by kripken — APPROVED** [review]

**Pull request merged**